### ISSUE TITLE
Buttons not pushing to Navigation Bar

### ISSUE THREAD

**Issue by dianevalanche** [photokandyStudios/PKNativeControls#3]
Based on <a href="https://github.com/photokandyStudios/PKNativeControls/issues/1">the previous issue here</a>, I'm able to create a navigation bar, create a navigation item with a title, and push it to the bar with no problems. However, when I try to add buttons, the bar just shows up blank. 

Here's what I'm working with:

<pre>
 document.addEventListener("deviceready", function(){
    var navigationBar = window.nativeControls.NavigationBar();
            navigationBar.tintColor = window.nativeControls.Color ( "red" );
            navigationBar.barTintColor = window.nativeControls.Color ( "white" );
            navigationBar.addToView();
            
        var navigationItem = window.nativeControls.NavigationItem();
                                  navigationItem.leftButtons = [button1, button2];
                                  navigationItem.rightButtons = [button3, button4];
                                  
                                  var button1 = window.nativeControls.BarButton();
                                  button1.title = "One";
                                  var button2 = window.nativeControls.BarButton();
                                  button2.title = "Two";
                                  var button3 = window.nativeControls.BarButton();
                                  button3.title = "Three";
                                  var button4 = window.nativeControls.BarButton();
                                  button4.title = "Four";

             navigationBar.push(navigationItem);
            });
</pre>

I've tried the buttons with text, images, and images + text, just in case – didn't change anything. I've tried giving NavigationItem both a title and buttons, or just buttons on one side – nothing. And again, the NavigationItem pushes just fine when it only has a title. I'm not sure what I'm missing – I'm very new at this, but this is pretty much direct from the documentation. Any insight is very much appreciated!

**Comment by rajdsouza** [comment]
You have defined your buttons after adding it to the navigation Bar.

Try the code below. 

```
document.addEventListener("deviceready", function(){
    var navigationBar = window.nativeControls.NavigationBar();
            navigationBar.tintColor = window.nativeControls.Color ( "red" );
            navigationBar.barTintColor = window.nativeControls.Color ( "white" );
            navigationBar.addToView();

        var navigationItem = window.nativeControls.NavigationItem();

                                  var button1 = window.nativeControls.BarButton();
                                  button1.title = "One";
                                  var button2 = window.nativeControls.BarButton();
                                  button2.title = "Two";
                                  var button3 = window.nativeControls.BarButton();
                                  button3.title = "Three";
                                  var button4 = window.nativeControls.BarButton();
                                  button4.title = "Four";

                                  navigationItem.leftButtons = [button1, button2];
                                  navigationItem.rightButtons = [button3, button4];

             navigationBar.push(navigationItem);
            });
```

**Comment by rajdsouza** [comment]
Also if you want to give custom colours to your Navigation bar. try the below in RGBA

```
navigationBar.barTintColor = {r:238,g:55,b:31,a:0};
```

**Comment by kerrishotts** [comment]
@rajdsouza Thanks; that was what I was going to suggest -- just hand't got there yet.

FYI; I've updated the documentation in the readme -- hopefully it's a little more helpful. I've moved the bar buttons section ahead of the navigation bar/item section in the hopes that someone writing as they read the documentation will define things by default, rather than having to guess at it. 

@dianevalanche Let me know if you have any additional problems or if the solution above doesn't help.
